pkgs/fixnum/test/int32_test.dart

```diff
@@ -214,18 +214,25 @@ void main() {
     });
 
     test('==', () {
-      expect(Int32(17), isNot(equals(Int32(18))));
-      expect(Int32(17), equals(Int32(17)));
-      expect(Int32(17), isNot(equals(Int32(16))));
-      expect(Int32(17), isNot(equals(Int64(18))));
-      expect(Int32(17), equals(Int64(17)));
-      expect(Int32(17), isNot(equals(Int64(16))));
-      expect(Int32.MIN_VALUE, isNot(equals(Int32.MAX_VALUE)));
-      expect(Int32(17), isNot(equals(18)));
-      expect(Int32(17) == 17, isTrue); // ignore: unrelated_type_equality_checks
-      expect(Int32(17), isNot(equals(16)));
-      expect(Int32(17), isNot(equals(Object())));
-      expect(Int32(17), isNot(equals(null)));
+      // Note: do not use `equals` matcher below as it considers exceptions as
```

Comment:
Should still be able to use it for equals, just not for `isNot(equals(...))`.

But I guess it's easier to just use the same approach everywhere.

Reply:
Even then, exceptions in `==` should still cause a mismatch in `equals`, so the test still fails and should throw a test failure, instead of the exception that `==` threw itself. (But `isNot(equals...)` won't throw, and that's a problem.)